modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/ServiceTemplateManage.java

```diff
@@ -91,36 +93,78 @@ public class ServiceTemplateManage {
+            }
+        }
+        ServiceTemplateEntity serviceTemplateToUpdate = new ServiceTemplateEntity();
+        BeanUtils.copyProperties(existingServiceTemplate, serviceTemplateToUpdate);
+        iconUpdate(serviceTemplateToUpdate, ocl);
+        checkParams(serviceTemplateToUpdate, ocl);
         validateRegions(ocl);
         validateFlavors(ocl);
+        validateServiceDeployment(ocl.getDeployment(), serviceTemplateToUpdate);
         billingConfigValidator.validateBillingConfig(ocl);
-        if (Objects.nonNull(existingTemplate.getOcl().getServiceConfigurationManage())
+        if (Objects.nonNull(serviceTemplateToUpdate.getOcl().getServiceConfigurationManage())
                 && Objects.nonNull(ocl.getServiceConfigurationManage())) {
             serviceConfigurationParameterValidator.validateServiceConfigurationParameters(ocl);
         }
-        existingTemplate.setIsUpdatePending(true);
-        ServiceTemplateEntity updatedServiceTemplate =
-                templateStorage.storeAndFlush(existingTemplate);
-        validateServiceDeployment(ocl.getDeployment(), updatedServiceTemplate);
-        updatedServiceTemplate.setOcl(ocl);
-        setServiceTemplateRegistrationState(updatedServiceTemplate);
-        updatedServiceTemplate.setIsUpdatePending(false);
-        updatedServiceTemplate = templateStorage.storeAndFlush(existingTemplate);
-        serviceTemplateOpenApiGenerator.updateServiceApi(updatedServiceTemplate);
-        return updatedServiceTemplate;
+        serviceTemplateToUpdate.setOcl(ocl);
+        boolean isAutoApprovedEnabled =
+                isAutoApproveEnabledForCsp(ocl.getCloudServiceProvider().getName());
+        final ServiceTemplateHistoryEntity storedUpdateHistory =
+                createServiceTemplateHistory(isAutoApprovedEnabled,
+                        ServiceTemplateRequestType.UPDATE, serviceTemplateToUpdate);
+        ServiceTemplateEntity updatedServiceTemplate;
+        // if auto approved is enabled by CSP, approved service template with new ocl directly
+        if (isAutoApprovedEnabled) {
+            updateServiceTemplateState(true, ServiceTemplateRequestType.UPDATE,
+                    serviceTemplateToUpdate);
+            updatedServiceTemplate = templateStorage.storeAndFlush(serviceTemplateToUpdate);
+        } else {
+            updateServiceTemplateState(false, ServiceTemplateRequestType.UPDATE,
+                    existingServiceTemplate);
+            if (existingServiceTemplate.getAvailableInCatalog()
+                    && isRemoveFromCatalogUntilApproved) {
+                existingServiceTemplate.setAvailableInCatalog(false);
+            }
+            updatedServiceTemplate = templateStorage.storeAndFlush(existingServiceTemplate);
+        }
+        // update service template openApi doc only when service template is available in catalog
+        if (updatedServiceTemplate.getAvailableInCatalog()) {
```

Comment:
here the value can be true but the update can be still in-review. So better to keep this as open-api generation as part of the block in line 139